pkg/cmd/experimental/syncgroups/ad/ldapinterface.go

```diff
@@ -94,13 +94,13 @@ func (e *ADLDAPInterface) ListGroups() ([]string, error) {
 		return nil, err
 	}
 
-	return sets.KeySet(reflect.ValueOf(e.ldapGroupToLDAPMembers)).List(), nil
+	return sets.KeySet(reflect.ValueOf(e.cachedGroupedUsers)).List(), nil
 }
 
 // populateCache queries all users to build a map of all the groups.  If the cache has already been
 // populated, this is a no-op.
 func (e *ADLDAPInterface) populateCache() error {
-	if e.cachePopulated {
+	if len(e.cachedGroupedUsers) > 0 {
```

Comment:
This check is **not** equivalent.  If you extract members first, then do a list, you've returned the wrong data.  In general I'm with refactors that are equivalent being combined, but refactors that are not equivalent shouldn't be hidden like this.